### PR TITLE
NIFI-11676 Added EL support to every text field in the StandardOauth2…

### PR DESCRIPTION
**Pull request by MikeThomsen**
…AccessToken

# Summary

[NIFI-00000](https://issues.apache.org/jira/browse/NIFI-00000)

# Tracking

Please complete the following tracking steps prior to pull request creation.

### Issue Tracking

- [ ] [Apache NiFi Jira](https://issues.apache.org/jira/browse/NIFI) issue created

### Pull Request Tracking

- [ ] Pull Request title starts with Apache NiFi Jira issue number, such as `NIFI-00000`
- [ ] Pull Request commit message starts with Apache NiFi Jira issue number, as such `NIFI-00000`

### Pull Request Formatting

- [ ] Pull Request based on current revision of the `main` branch
- [ ] Pull Request refers to a feature branch with one commit containing changes

# Verification

Please indicate the verification steps performed prior to pull request creation.

### Build

- [ ] Build completed using `mvn clean install -P contrib-check`
  - [ ] JDK 17

### Licensing

- [ ] New dependencies are compatible with the [Apache License 2.0](https://apache.org/licenses/LICENSE-2.0) according to the [License Policy](https://www.apache.org/legal/resolved.html)
- [ ] New dependencies are documented in applicable `LICENSE` and `NOTICE` files

### Documentation

- [ ] Documentation formatting appears as expected in rendered files

### CODE REVIEW

File: nifi-nar-bundles/nifi-standard-services/nifi-oauth2-provider-bundle/nifi-oauth2-provider-service/src/main/java/org/apache/nifi/oauth2/StandardOauth2AccessTokenProvider.java [hunk 1]
```diff
@@ -125,6 +125,7 @@ public class StandardOauth2AccessTokenProvider extends AbstractControllerService
         .required(true)
         .sensitive(true)
         .addValidator(StandardValidators.NON_BLANK_VALIDATOR)
+        .expressionLanguageSupported(ExpressionLanguageScope.VARIABLE_REGISTRY)
```
Comment (exceptionfactory) [inline nifi-nar-bundles/nifi-standard-services/nifi-oauth2-provider-bundle/nifi-oauth2-provider-service/src/main/java/org/apache/nifi/oauth2/StandardOauth2AccessTokenProvider.java:128]: Sensitive properties should not support Expression Language.
```suggestion
        .expressionLanguageSupported(ExpressionLanguageScope.NONE)
```

File: nifi-nar-bundles/nifi-standard-services/nifi-oauth2-provider-bundle/nifi-oauth2-provider-service/src/main/java/org/apache/nifi/oauth2/StandardOauth2AccessTokenProvider.java [hunk 2]
```diff
@@ -153,6 +154,7 @@ public class StandardOauth2AccessTokenProvider extends AbstractControllerService
...
```
Comment (exceptionfactory) [inline nifi-nar-bundles/nifi-standard-services/nifi-oauth2-provider-bundle/nifi-oauth2-provider-service/src/main/java/org/apache/nifi/oauth2/StandardOauth2AccessTokenProvider.java:157]:
```suggestion
        .expressionLanguageSupported(ExpressionLanguageScope.NONE)
```

File: nifi-nar-bundles/nifi-standard-services/nifi-oauth2-provider-bundle/nifi-oauth2-provider-service/src/main/java/org/apache/nifi/oauth2/StandardOauth2AccessTokenProvider.java [hunk 3]
```diff
@@ -258,12 +264,12 @@ public void onEnabled(ConfigurationContext context) {
         clientAuthenticationStrategy = ClientAuthenticationStrategy.valueOf(context.getProperty(CLIENT_AUTHENTICATION_STRATEGY).getValue());
         grantType = context.getProperty(GRANT_TYPE).getValue();
         username = context.getProperty(USERNAME).evaluateAttributeExpressions().getValue();
-        password = context.getProperty(PASSWORD).getValue();
+        password = context.getProperty(PASSWORD).evaluateAttributeExpressions().getValue();
```
Comment (exceptionfactory) [inline nifi-nar-bundles/nifi-standard-services/nifi-oauth2-provider-bundle/nifi-oauth2-provider-service/src/main/java/org/apache/nifi/oauth2/StandardOauth2AccessTokenProvider.java:267]:
```suggestion
        password = context.getProperty(PASSWORD).getValue();
```

File: nifi-nar-bundles/nifi-standard-services/nifi-oauth2-provider-bundle/nifi-oauth2-provider-service/src/test/java/org/apache/nifi/oauth2/StandardOauth2AccessTokenProviderTest.java [hunk 4]
```diff
@@ -120,13 +123,13 @@ protected ComponentLog getLogger() {
         when(mockContext.getProperty(StandardOauth2AccessTokenProvider.GRANT_TYPE).getValue()).thenReturn(StandardOauth2AccessTokenProvider.RESOURCE_OWNER_PASSWORD_CREDENTIALS_GRANT_TYPE.getValue());
         when(mockContext.getProperty(StandardOauth2AccessTokenProvider.AUTHORIZATION_SERVER_URL).evaluateAttributeExpressions().getValue()).thenReturn(AUTHORIZATION_SERVER_URL);
         when(mockContext.getProperty(StandardOauth2AccessTokenProvider.USERNAME).evaluateAttributeExpressions().getValue()).thenReturn(USERNAME);
-        when(mockContext.getProperty(StandardOauth2AccessTokenProvider.PASSWORD).getValue()).thenReturn(PASSWORD);
+        when(mockContext.getProperty(StandardOauth2AccessTokenProvider.PASSWORD).evaluateAttributeExpressions().getValue()).thenReturn(PASSWORD);
```
Comment (exceptionfactory) [inline nifi-nar-bundles/nifi-standard-services/nifi-oauth2-provider-bundle/nifi-oauth2-provider-service/src/test/java/org/apache/nifi/oauth2/StandardOauth2AccessTokenProviderTest.java:126]:
```suggestion
        when(mockContext.getProperty(StandardOauth2AccessTokenProvider.PASSWORD).getValue()).thenReturn(PASSWORD);
```

**Review by exceptionfactory — CHANGES_REQUESTED** [review]
Thanks for the improvement @MikeThomsen. I noted that sensitive properties should not support Expression Language, so those several references should be removed.

Although the Variable Registry concept is deprecated for removal, this approach will still support environment variables or system properties, so it seems viable within that scope. Alternatively, these values could be provided through Parameter Contexts, and one of several Parameter Providers. Is there a particular reason for supporting Expression Language in this case?

**Comment by MikeThomsen** [comment]
@exceptionfactory based on [the ticket](https://issues.apache.org/jira/browse/NIFI-11676) I assumed that the user was having issues with accessing parameters because EL was not supported. Did I miss something on the underlying implementation?

**Comment by exceptionfactory** [comment]
@MikeThomsen On further review of the associated Jira issue, I think this may actually be a rendering issue in the UI. Support for Parameters is not related to support for Expression Language, and the UI drives the supported capabilities based on several other factors. There were some recent changes for [NIFI-11287](https://issues.apache.org/jira/browse/NIFI-11287) related to referencing parameters, so there might be some closer evaluation necessary. Either way, however, the Expression Language support changes in this PR are not related support for Parameters, so it seems best to close this PR and take a closer look at the recent UI changes.

**Pull request closed without merging**